.github/workflows/gradle-precommit.yml

```diff
@@ -19,5 +19,8 @@ jobs:
         java-version: 11
     - name: Grant execute permission for gradlew
       run: chmod +x gradlew
+    - name: Run something to download the gradle wrapper jar
```

Comment:
This has to work out of the box for ./gradlew precommit. You shouldn't need to run anything prior to that.

Reply:
It does work out of the box. The reason that I added an intermediate step was so that we can run the validator before running the rest of precommit.

Reply:
I don't see the point. It either will download gradle jar or it won't (and fail) before it runs precommit.

Reply:
validator will verify the checksum, which I felt was important before we go on to running the rest of precommit. Maybe my threat model is wrong here? We can move validator action to after precommit in that case?

Reply:
The checksum should be checked as part of the script that downloads the wrapper jar... otherwise it's too late and you've executed something you shouldn't have. That's what I meant when I said it adds additional complexity...

Reply:
Oh.. now I see which validator you had in mind. Sorry for being dim.